auth/gcloud/aio/auth/iam.py

```diff
@@ -5,23 +5,29 @@
 from typing import Optional
 from typing import Union
 
-import aiohttp
-
+from .build_constants import BUILD_GCLOUD_REST
+from .session import AioSession as RestSession
 from .token import Token
 from .token import Type
 from .utils import encode
 
+# Selectively load libraries based on the package
+# TODO: Can we somehow just pick up the pacakge name instead of this
+if BUILD_GCLOUD_REST:
+    from requests import Session
+else:
+    from aiohttp import ClientSession as Session
```

Comment:
Hmm, I do like the explicitness of this block. You're going to get async unless you explicitly tell the library to be built as synchronous.

Reply:
Fair enough -- I added a bunch of comments throughout this PR about ways we could avoid the `BUILD_GCLOUD_REST` variable, since a build var which gets magically changed by a `sed` command might be a bit hard to grok; not all places are necessarily better without this block... but I think it might be worth it to remove that key; the fewer places we need to think through building multiple code paths to add a feature to `gcloud-aio` / `gcloud-rest`, the better; otherwise we're not going to get too much of a win from merging the codebases, since we'll be building things twice anyway.

That said, maybe there are some places where the `if` makes more sense! I could go either way in some cases.

Reply:
So, I think we can keep this while we are importing modules. Because depending on the client's `pyenv` it might have `aiohttp` etc. if they are using `gcloud-rest` in `python3`. Then we can get weird issues when wrong classes are being loaded.

However, I agree, we can remove the `if`s that are inline in code that are handling incompatibilities in package APIs.